src/charm.py

```diff
@@ -746,14 +746,18 @@ def restart_mongod_service(self, auth=None):
         """Restarts the mongod service with its associated configuration."""
         if auth is None:
             auth = self.auth_enabled()
-
-        stop_mongod_service()
-        update_mongod_service(
-            auth,
-            self._unit_ip(self.unit),
-            config=self.mongodb_config,
-        )
-        start_mongod_service()
+        try:
+            stop_mongod_service()
+            update_mongod_service(
+                auth,
+                self._unit_ip(self.unit),
+                config=self.mongodb_config,
+            )
+            start_mongod_service()
+            return True
+        except systemd.SystemdError as e:
```

Comment:
This method is always called as a void. So I think the `return True` is unnecessary.
Also, the exception is not handled anywhere. It's explicit here, but if it wasn't, it would raise regardless and show on the `juju-debug` log.
Also also, I think you can remove `restart_mongod_service` from `src/machine_helpers.py`?

Reply:
great feedback thank you!